File: iOS/PengChauFerryTests/FerryDayTest.swift
//
//  FerryDayTest.swift
//  PengChauFerryTests
//
//  Created by Lionello Lunesu on 8/3/2020.
//  Copyright © 2020 Lionello Lunesu. All rights reserved.
//

import XCTest
@testable import PengChauFerry

class FerryDayTest: XCTestCase {

    func testEveryday() {
        XCTAssertEqual(8, FerryDay.intToDays(255).count)
    }

    func testToday() {
        let weekdays = FerryDay.intToDays(127)
        XCTAssertEqual(true, weekdays.contains(FerryDay.today()))
    }

    func testIntAndBack() {
        for i in 1...255 {
            XCTAssertEqual(UInt8(i), FerryDay.daysToInt(FerryDay.intToDays(UInt8(i))))
        }
    }

    func testToString() {
        XCTAssertEqual("Monday", FerryDay.Monday.toString())
        XCTAssertEqual("Tuesday", FerryDay.Tuesday.toString())
        XCTAssertEqual("Wednesday", FerryDay.Wednesday.toString())
        XCTAssertEqual("Thursday", FerryDay.Thursday.toString())
        XCTAssertEqual("Friday", FerryDay.Friday.toString())
        XCTAssertEqual("Saturday", FerryDay.Saturday.toString())
        XCTAssertEqual("Sunday", FerryDay.Sunday.toString())
        XCTAssertEqual("Holiday", FerryDay.Holiday.toString())
    }

}

File: Android/app/src/test/java/com/lunesu/pengchauferry/FerryDayTest.kt
package com.lunesu.pengchauferry

import org.junit.Assert.assertEquals
import org.junit.Test

class FerryDayTest {
    @Test
    fun testEveryday() {
        assertEquals(8, FerryDay.intToDays(255).size)
    }

    @Test
    fun testToday() {
        val weekdays = FerryDay.intToDays(127)
        assertEquals(true, weekdays.contains(FerryDay.today()))
    }

    @Test
    fun testIntAndBack() {
        for (i in 1..255) {
            assertEquals(i, FerryDay.daysToInt(FerryDay.intToDays(i)))
        }
    }

    @Test
    fun testToString() {
        assertEquals("Monday", FerryDay.Monday.toString())
        assertEquals("Tuesday", FerryDay.Tuesday.toString())
        assertEquals("Wednesday", FerryDay.Wednesday.toString())
        assertEquals("Thursday", FerryDay.Thursday.toString())
        assertEquals("Friday", FerryDay.Friday.toString())
        assertEquals("Saturday", FerryDay.Saturday.toString())
        assertEquals("Sunday", FerryDay.Sunday.toString())
        assertEquals("Holiday", FerryDay.Holiday.toString())
    }
}
